debezium-embedded/src/main/java/io/debezium/embedded/OffsetCommitPolicy.java

```diff
@@ -74,4 +96,8 @@ default OffsetCommitPolicy and(OffsetCommitPolicy other) {
         if ( other == null ) return this;
         return (number, time, unit) -> this.performCommit(number, time, unit) && other.performCommit(number, time, unit);
     }
+
+    default OffsetCommitPolicy configure(Configuration config) {
```

Comment:
This prevents to make any config options retrieved from `Configuration` to be stored in final fields. How about enhancing the `getInstance()` logic in the config framework so that we either accept no-args constructors or constructors with a single parameter of type `Configuration` (preferring that one in case both are present). That way custom implementors can fetch any config and safely assign it in a final field.